modules/haze_removal/include/opencv2/haze_removal/dark_channel_algorithm.hpp

```diff
@@ -0,0 +1,48 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef OPENCV_HAZE_REMOVAL_HE_ALGORITHM_HPP
+#define OPENCV_HAZE_REMOVAL_HE_ALGORITHM_HPP
+
+#include "haze_removal_base.hpp"
+
+namespace cv {
+namespace haze_removal {
+
+//! @addtogroup haze_removal
+//! @{
+
+/** @brief Dehazes a image using Dark Channel Prior
+
+For details refer to @cite kaiminghe_cvpr09
+*/
+class CV_EXPORTS_W DarkChannelPriorHazeRemoval : public HazeRemovalBase
+{
+public:
+    CV_WRAP void setKernel(int _erosionSize, int _erosionType);
+    CV_WRAP void setKernel(cv::InputArray _kernelForEroding);
```

Comment:
Please avoid `cv::` prefixes in public API headers

Reply:
Fixed.